doc/api/assert.md

```diff
@@ -767,17 +767,39 @@ assert.throws(
 
 Note that `error` can not be a string. If a string is provided as the second
 argument, then `error` is assumed to be omitted and the string will be used for
-`message` instead. This can lead to easy-to-miss mistakes:
+`message` instead. This can lead to easy-to-miss mistakes. Please read the
+example below carefully if you intend to use a string as second argument:
 
 <!-- eslint-disable no-restricted-syntax -->
 ```js
-// THIS IS A MISTAKE! DO NOT DO THIS!
-assert.throws(myFunction, 'missing foo', 'did not throw with expected message');
-
-// Do this instead.
-assert.throws(myFunction, /missing foo/, 'did not throw with expected message');
+function throwingMismatch() {
+  throw new Error('Mismatch');
+}
+function throwingMatched() {
+  throw new Error('Matched');
+}
+function notThrowing() {}
+
+// The second argument is a string and the input function threw an Error.
+// In that case both cases do not throw as neither is going to try to
+// match for the error message thrown by the input function!
+assert.throws(throwingMismatch, 'Error: Matched');
+assert.throws(throwingMatched, 'Error: Matched');
+
+// The string is only used (as message) in case the function does not throw:
+assert.throws(notThrowing, 'Error: Matched');
+// AssertionError [ERR_ASSERTION]: Missing expected exception: Matched
+
+// If you originally intended to match for the error message do this instead:
+assert.throws(throwingMatched, /^Error: Matched$/);
+// Does not throw because the error messages match.
+assert.throws(throwingMismatch, /^Error: Matched$/);
```

Comment:
Am I missing something subtle? It looks like line 794 and line 796 are the same?

Reply:
The called function is named similar but different. I clarified it now by choosing a different name.